books/migrations/0003_change_author_table_&apply_data_migration.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 3.2.5 on 2021-07-13 20:21
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def populate_long_names(apps, schema_editor):
+    Author = apps.get_model('books', 'Author')
+    for author in Author.objects.all():
+        author.short_name = '%s' % (author.first_name)
+        author.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('books', '0002_rename_publication_year_book_publication_date'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='author',
+            name='short_name',
+            field=models.CharField(max_length=200, null=True),
+        ),
+        migrations.RunPython(populate_long_names),
```

Comment:
move this at the end of scheme alterations.

Reply:
i kept it here because i am populating the short_name from first_name which is later getting deleted.

Reply:
basically, i am deleting the old fields first_name and last_name from authors models and creating 3 new ones, short_name, area_of_interest and user. In order to pre-populate previous rows in this manner, I am setting the value of first_name to short_name and later on removing the field from the model. Hence, the order. let me know if i need to change it still

Reply:
I see, in that case, I would recommend moving `short_name` to the user's table because it's a user's property rather than the authors property.

Reply:
these fields were suggested by tehreem. I dont know what she had in mind. are you sure i should make this change?

Reply:
No, lets leave it here.

Reply:
sure